tck/base/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingClassMethodNameClientTests.java

```diff
@@ -50,6 +51,11 @@ public String getValue(String s) {
         public String getName() {
             return this.getClass().getName();
         }
+
+        @Override
+        public Set<String> getPropertyNames() {
```

Comment:
You should not rely on getProperties as you will run into deadlock unless you implement getProperties method yourself.

Reply:
ah. checked that you implemented the getProperties().